### ISSUE TITLE
torch.nn.functional.binary_cross_entropy and torch.nn.functional.binary_cross_entropy_with_logits documentation wrong on description about target 

### ISSUE THREAD

**Issue by cheyennee** [pytorch/pytorch#99151]
### 📚 The doc issue

The [binary_cross_entropy documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy.html) shows that `target` – Tensor of the same shape as input with values between 0 and 1. However, the value of `target`  does not necessarily have to be between 0-1, but the value of `input` must be between 0-1.
```
import torch
arg_1_tensor = torch.randint(2, 3, [5, 5], dtype=torch.float32)
arg_2_tensor = torch.randint(2, 3, [5, 5], dtype=torch.float32)
arg_3 = None
arg_4 = "mean"
res = torch.nn.functional.binary_cross_entropy(input=arg_1_tensor,target=arg_2_tensor,weight=arg_3,reduction=arg_4,)
print(res)
# res: RuntimeError: all elements of input should be between 0 and 1
```

The [binary_cross_entropy_with_logits documentation](https://pytorch.org/docs/stable/generated/torch.nn.functional.binary_cross_entropy_with_logits.html) shows that `target` – Tensor of the same shape as input with values between 0 and 1. However, the value of `target`  does not necessarily have to be between 0-1.
```
import torch
arg_1_tensor = torch.randint(2, 4, [10, 64], dtype=torch.float32)
arg_2_tensor = torch.randint(2, 4, [10, 64], dtype=torch.float32)
arg_3 = None
arg_4_tensor = torch.rand([64], dtype=torch.float32)
arg_4 = arg_4_tensor.clone()
arg_5 = "mean"
res = torch.nn.functional.binary_cross_entropy_with_logits(input=arg_1_tensor,target=arg_2_tensor,weight=arg_3,pos_weight=arg_4,reduction=arg_5,)
print(res)
# res: tensor(-3.8518)
```

cc @svekars @carljparker @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by MylesJP** [comment]
Mind if I take a shot at this one for a school project?

**Comment by cheyennee** [comment]
Sure！



| |
17362995059
|
|
***@***.***
|




---- 回复的原邮件 ----
| 发件人 | Myles ***@***.***> |
| 日期 | 2023年04月15日 00:07 |
| 收件人 | ***@***.***> |
| 抄送至 | ***@***.***>***@***.***> |
| 主题 | Re: [pytorch/pytorch] torch.nn.functional.binary_cross_entropy and torch.nn.functional.binary_cross_entropy_with_logits documentation wrong on description about target (Issue #99151) |

Mind if I take a shot at this one for a school project?

**Comment by jbschlosser** [comment]
IIRC we've tried to introduce a [0, 1] restriction for `target` in the past but ran into BC issues from people using values outside the range. I'm not sure we want to encourage this by changing the docs, however.

**Comment by jbschlosser** [comment]
I take it back; it looks like this was addressed and actually landed in #97814. Closing as addressed but feel free to reopen if there's more I missed.

**Comment by cheyennee** [comment]
> Mind if I take a shot at this one for a school project?

Sure!
